full/monolith/folders.tf

```diff
@@ -1,64 +1,35 @@
-resource "google_folder" "common" {
-  display_name = "Common ${random_id.suffix.hex}"
-  parent       = "organizations/${var.org_id}"
-}
-
-resource "google_folder" "team-1" {
-  display_name = "Team 1 ${random_id.suffix.hex}"
-  parent       = "organizations/${var.org_id}"
-}
-
-resource "google_folder" "team-1-development" {
-  display_name = "Development ${random_id.suffix.hex}"
-  parent       = google_folder.team-1.name
-}
-
-resource "google_folder" "team-1-non-production" {
-  display_name = "Non-Production ${random_id.suffix.hex}"
-  parent       = google_folder.team-1.name
-}
-
-resource "google_folder" "team-1-production" {
-  display_name = "Production ${random_id.suffix.hex}"
-  parent       = google_folder.team-1.name
-}
-
-resource "google_folder" "team-2" {
-  display_name = "Team 2 ${random_id.suffix.hex}"
-  parent       = "organizations/${var.org_id}"
-}
-
-resource "google_folder" "team-2-development" {
-  display_name = "Development ${random_id.suffix.hex}"
-  parent       = google_folder.team-2.name
-}
 
-resource "google_folder" "team-2-non-production" {
-  display_name = "Non-Production ${random_id.suffix.hex}"
-  parent       = google_folder.team-2.name
-}
+module "common" {
+  source  = "terraform-google-modules/folders/google"
+  version = "~> 3.0"
 
-resource "google_folder" "team-2-production" {
-  display_name = "Production ${random_id.suffix.hex}"
-  parent       = google_folder.team-2.name
+  parent = "organizations/${var.org_id}"
+  names = [
+    "Common",
+  ]
 }
 
-resource "google_folder" "team-3" {
-  display_name = "Team 3 ${random_id.suffix.hex}"
-  parent       = "organizations/${var.org_id}"
-}
+module "teams" {
+  source  = "terraform-google-modules/folders/google"
+  version = "~> 3.0"
 
-resource "google_folder" "team-3-development" {
-  display_name = "Development ${random_id.suffix.hex}"
-  parent       = google_folder.team-3.name
+  parent = "organizations/${var.org_id}"
+  names = [
+    "Team 1",
+    "Team 2",
+    "Team 3"
+  ]
 }
 
-resource "google_folder" "team-3-non-production" {
-  display_name = "Non-Production ${random_id.suffix.hex}"
-  parent       = google_folder.team-3.name
-}
+module "envs" {
+  for_each = module.teams.ids
+  source   = "terraform-google-modules/folders/google"
+  version  = "~> 3.0"
 
-resource "google_folder" "team-3-production" {
-  display_name = "Production ${random_id.suffix.hex}"
-  parent       = google_folder.team-3.name
+  parent = each.value
```

Comment:
Should this be the other way around? i.e. top-level folder for dev/non-prod/prod and then team folders underneath? That makes centralized env specific configurations easier at the folder level.

Reply:
The hierarchies are mostly based off what we identified for LZ in https://github.com/GoogleCloudPlatform/blueprints/tree/main/catalog/hierarchy. 
I suspect we will need to support flexibility either way on FS side which will create these dynamically. Since this is a refactor PR, I will hold off on making any architecture change.